cli/internal/cmd/upgradeapply.go

```diff
@@ -413,6 +426,34 @@ func (u *upgradeApplyCmd) handleServiceUpgrade(cmd *cobra.Command, conf *config.
 	return err
 }
 
+// migrateFrom2_10 applies migrations necessary for upgrading from v2.10 to v2.11
+// TODO(v2.11): Remove this function after v2.11 is released.
+func migrateFrom2_10(ctx context.Context, kubeUpgrader kubernetesUpgrader) error {
+	// Sanity check to make sure we only run migrations on upgrades with CLI version 2.10 < v < 2.12
+	if !constants.BinaryVersion().MajorMinorEqual(semver.NewFromInt(2, 11, 0, "")) {
```

Comment:
Wouldn't it be more robust to check the release version that is currently in the cluster?

Reply:
This check was mostly meant as a sanity check so we don't run unnecessary migrations if we forget to remove code.
In case we want a more general approach to applying migration during `upgrade apply` with extra safety checks etc., I am open to discuss and implement them, but I would like to keep that separate from this PR.